### PR TITLE
Feature/310 cookie consent

### PR DESCRIPTION
**Pull request by patric-eberle**
## Pull request
This PR adds the cookiebot integration from one of our projects, so content can be blocked, until the consent has been given. Since we don't have a general cookiebot license, the page is not working before a valid instance was added to the page. I hope this is clear enough by the code comments.

Also note: the many file changes come from the fact, that I introduced nesting folders for the styleguide pages, like we d in generalo for every project.
 
### Ticket
#310 
 
### Browser testing
- http://localhost:5173/styleguide/components/cookie-consent
 
### Checklist
- [x] I merged the current development branch (before testing)
- [x] Added JSDoc and styleguide demo
- [x] Tested all links in project relevant browsers
- [x] Tested all links in different screen sizes
- [x] Did run automated tests and linters
- [x] Did assign ticket
- [x] Double checked target branch
 
## Review/Test checklist
- [ ] Did review code and documentation
- [ ] Tested all links in project relevant browsers
- [ ] Tested all links in different screen sizes
- [ ] Did check accessibility (Wave, only errors)
- [ ] Re-assign ticket to developer

### CODE REVIEW

File: src/plugins/consent/c-consent-gatekeeper.vue [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+    name: 'c-consent-gatekeeper',
+    components: {
+      eButton,
+    },
+
+    // components: {},
+
+    props: {
+      /**
+       * Expects a consent group identifier.
```
Comment (BeneRichi) [inline src/plugins/consent/c-consent-gatekeeper.vue:41]: question: Why use 'identifier'? Isn't it the entire consent group?

**Comment by patric-eberle** [comment]
@BeneRichi I updated the comment wording. Can you please check again? THX.

File: src/styleguide/routes/components/r-consent.vue [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    // mounted() {},
+    // beforeUpdate() {},
+    // updated() {},
+    // activated() {},
+    // deactivated() {},
+    // beforeUnmount() {},
+    // unmounted() {},
+
+    methods: {
+      onConsentApprove() {
```
Comment (BeneRichi) [inline src/styleguide/routes/components/r-consent.vue:60]: suggestion: Add `:void` as return type.

**Comment by BeneRichi** [comment]
Great job! To align with our TypeScript approach, please take a look at his [comment](https://github.com/valantic/vue-template/pull/348#discussion_r1448632927). After that, you can merge.

**Review by BeneRichi — APPROVED** [review]

**Pull request merged**